commit cd32d6553e27c4b23e2dfba98e14d88fbd1c9ca1
Author: Daniel Bermond <danielbermond@gmail.com>
Date:   Fri Sep 27 17:19:54 2019 +0000

    Makefile: fix webui target
    
    The webui target is not a file. This causes the webui
    target and all targets that depends on it to be always
    rebuilded, which is not desirable.
    
    Most notably, this causes a second compilation at
    'make install' when using the traditional approach
    to build software with 'make && make install', which
    should not occur.
    
    This commit fixes this situation and now 'make install'
    only install files if they were already built.
    
    Signed-off-by: Daniel Bermond <danielbermond@gmail.com>

diff --git a/Makefile b/Makefile
index 1329108..fdb90f9 100644
--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,8 @@ WEB_FILES = $(DOCROOT)/index.html
 WEB_FILES += $(DOCROOT)/xd.min.js
 WEB_FILES += $(DOCROOT)/xd.css
 WEB_FILES += $(WEBUI_LOGO)
+WEBUI_CORE  = $(DOCROOT)/xd.min.js
+WEBUI_CORE += $(DOCROOT)/xd.css
 WEBUI_PREFIX = /contrib/webui/docroot
 ASSETS = $(REPO)/src/xd/lib/rpc/assets/assets.go
 
@@ -51,7 +53,7 @@ assets: $(ASSETS)
 $(GO_ASSETS):
 	GOPATH=$(REPO) $(GO) build -o $(GO_ASSETS) -v github.com/jessevdk/go-assets-builder
 
-$(ASSETS): $(GO_ASSETS) webui
+$(ASSETS): $(GO_ASSETS) $(WEBUI_CORE)
 	$(GO_ASSETS) -p assets -s $(WEBUI_PREFIX) -o $(ASSETS) $(WEB_FILES)
 
 $(XD): $(ASSETS)
@@ -86,8 +88,10 @@ go-clean:
 $(WEBUI_LOGO):
 	$(CP) $(LOGOS)/xd_logo.png $(WEBUI_LOGO)
 
-webui: $(WEBUI_LOGO)
-	$(MAKE) -C $(WEBUI)	
+$(WEBUI_CORE): $(WEBUI_LOGO)
+	$(MAKE) -C $(WEBUI)
+
+webui: $(WEBUI_CORE)
 
 no-webui:
 	GOPATH=$(REPO) $(GO) build -ldflags "-X xd/lib/version.Git=$(GIT_VERSION)" -o $(XD)
